### ISSUE TITLE
Questions replaced by login pin

### ISSUE THREAD

**Issue by hearsid** [deCodeIt/Zerodha-Selenium-Automation#1]
Zerodha has replaced the questions by asking for a pin to login.

**Comment by deCodeIt** [comment]
Yeah, I never took this project to completion.
Let me see if I can take out time to fix this otherwise you can fork it and do it yourself

**Comment by hearsid** [comment]
I already have the working code for it and for additional stuff, but my concern is do Zerodha allow for selenium use? If they detect automation will they ban the account? (Since they have their own API which is for Rs.2000/mo)

**Comment by nav886** [comment]
> I already have the working code for it and for additional stuff, but my concern is do Zerodha allow for selenium use? If they detect automation will they ban the account? (Since they have their own API which is for Rs.2000/mo)

Could you please share this working code? I can’t seem to make it work. Every time I enter the pin, it is deleting 2 characters.

**Comment by deCodeIt** [comment]
Kite Zerodha now has GTT orders so do you really need this script? I think GTT works pretty good.

**Comment by hearsid** [comment]
@nav886 Here it is, please replace the sleep with waitsFor if you need faster login.
```
def login():
    browser.get('https://kite.zerodha.com/')
    time.sleep(2.0)
    browser.find_elements_by_css_selector('input')[0].send_keys(kite_login['username'])
    browser.find_element_by_css_selector(login_po['PASSWORD']).send_keys(kite_login['password'])  
    browser.find_element_by_css_selector(login_po['SUBMIT']).click()

    # on the next page enter the pin
    time.sleep(1.0)
    browser.find_element_by_css_selector(login_po['PIN']).send_keys(kite_login['pin'])
    browser.find_element_by_css_selector(login_po['SUBMIT']).click()
```

**Comment by hearsid** [comment]
@deCodeIt GTT have minimum upper and lower limits, so you can't do scalping or similar kind of trades.

**Comment by nav886** [comment]
> @nav886 Here it is, please replace the sleep with waitsFor if you need faster login.
> 
> ```
> def login():
>     browser.get('https://kite.zerodha.com/')
>     time.sleep(2.0)
>     browser.find_elements_by_css_selector('input')[0].send_keys(kite_login['username'])
>     browser.find_element_by_css_selector(login_po['PASSWORD']).send_keys(kite_login['password'])  
>     browser.find_element_by_css_selector(login_po['SUBMIT']).click()
> 
>     # on the next page enter the pin
>     time.sleep(1.0)
>     browser.find_element_by_css_selector(login_po['PIN']).send_keys(kite_login['pin'])
>     browser.find_element_by_css_selector(login_po['SUBMIT']).click()
> ```

Thanks for posting the code. Could you please let me know what login_po is?

**Comment by hearsid** [comment]
@nav886 Was out of town so couldn't reply earlier, here is the login_po (page object).
PAGE_OBJECT = {
    'USERNAME' : 'input[placeholder="User ID"]',
    'PASSWORD' : 'input[type="password"]',
    'PIN' : 'input[label="PIN"]',
    'SUBMIT' : 'button[type="submit"]'
}

**Comment by deCodeIt** [comment]
Fixed with PR 6e5b973
